### PR TITLE
[rllib] [rfc] add contrib module and guideline for merging

### PR DESCRIPTION
**Pull request by ericl**
## What do these changes do?

This adds guidelines for merging code into `rllib/contrib` vs `rllib/agents`. Also, clean up the agent import code to make registration easier.

### CODE REVIEW

File: doc/source/rllib-dev.rst [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+These are the guidelines for merging new algorithms into RLlib:
+
+* Contributed algorithms (`rllib/contrib <https://github.com/ray-project/ray/tree/master/python/ray/rllib/contrib>`__):
+    - must subclass Agent and implement the ``_train()`` method
+    - must include a lightweight test (<30s to run) to sanity check functionality
+    - should include tuned hyperparameter examples and documentation
+    - should offer functionality not present in existing algorithms
+
+* Fully integrated algorithms (`rllib/agents <https://github.com/ray-project/ray/tree/master/python/ray/rllib/agents>`__) have the following additional requirements:
```
Comment (robertnishihara) [inline doc/source/rllib-dev.rst]: Do we want contributions of *non*-fully integrated algorithms?

File: doc/source/rllib-dev.rst [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+Second, register the agent with a name in `contrib/registry.py <https://github.com/ray-project/ray/blob/master/python/ray/rllib/contrib/registry.py>`__.
+
+.. code-block:: python
+
+    def _import_random_agent():
+        from ray.rllib.contrib.random_agent.random_agent import RandomAgent
+        return RandomAgent
+
+    CONTRIBUTED_ALGORITHMS = {
+        "contrib/RandomAgent": _import_random_agent,
```
Comment (richardliaw) [inline doc/source/rllib-dev.rst]: on the docs, show two agents (is currently confusing as is)

File: doc/source/rllib-dev.rst [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    }
+
+After registration, you can run and visualize agent progress using ``rllib train``:
+
+.. code-block:: bash
+
+    rllib train --run=contrib/RandomAgent --env=CartPole-v0
+    tensorboard --logdir=~/ray_results
```
Comment (richardliaw) [inline doc/source/rllib-dev.rst]: should add a link for questions

**Review by richardliaw — APPROVED** [review]

**Comment by richardliaw** [comment]
(will merge after lint)

**Pull request merged**